distribution/docker/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM maven:3-jdk-8 as builder
+
+COPY . /src
+WORKDIR /src
+RUN mvn install -ff -DskipTests -Dforbiddenapis.skip=true -Pdist -Pbundle-contrib-exts
+RUN \
+ VER=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec) \
+ && tar -zxf ./distribution/target/apache-druid-${VER}-bin.tar.gz -C /opt \
+ && ln -s /opt/apache-druid-${VER} /opt/druid
+
+RUN cd /opt/druid/extensions/mysql-metadata-storage \
+ && wget -O mysql-connector-java-5.1.38.jar http://central.maven.org/maven2/mysql/mysql-connector-java/5.1.38/mysql-connector-java-5.1.38.jar \
```

Comment:
This might cause issues. Not sure what the limits of GPL stuff are, but having a script to download GPL components may or may not fly with Apache guidelines, will need confirmation.

Reply:
If this can be added as another (optional) layer that will probably help things out

Reply:
it is its own layer, but i don't really think it makes any difference. I'm not sure how i would make it optional, a sep build arg? a sep dockerfile?
the linux components (e.g. bash) are already gpl, so its not possible to have a container which doesn't have gpl.
and the gpl allows aggregation like this without attaching.
the other apache containers (e.g. tomcat, httpd, maven) all have gpl components in them.